### PR TITLE
Add gitlab provider

### PR DESCRIPTION
**Pull request by B3rs**
Add gitlab as project provider.

The gitlab provider can be used as follows
```yaml
  gitlab:
    token: # can also use $GITLAB_TOKEN
    url: 'https://gitlab.example.com/api/v4'
    limit: 10 #limits the number of results
    topics: 
      - something # searches gitlab projects with the specific topic set
    base-path: ./tmp/gitlab
    search: standards-insights #filter projects based on the search string
```

Every gitlab topic (formerly known as tags) that can be split using `: ` will be exported as a project label

### CODE REVIEW

**Comment by victorboissiere** [comment]
#sre

File: internal/providers/root.go [hunk 1]
```diff
@@ -18,6 +19,13 @@ func NewProviders(logger *slog.Logger, config config.ProvidersConfig) ([]aggrega
 		}
 		result = append(result, argoProvider)
 	}
+	if config.Gitlab.URL != "" {
+		argoProvider, err := gitlab.New(logger, config.Gitlab)
```
Comment (victorboissiere) [inline internal/providers/root.go:23]: wrong copy paste for the name var?

Comment (B3rs) [reply internal/providers/root.go:23]: thanks! fixed :)

File: internal/providers/gitlab/root.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		topic := strings.Join(c.config.Topics, ",")
+		options.Topic = &topic
+	}
+
+	if c.config.Search != "" {
+		options.Search = &c.config.Search
+	}
+
+	c.logger.Debug("fetching Gitlab projects")
+	projects, _, err := c.client.Projects.ListProjects(
```
Comment (mcorbin) [inline internal/providers/gitlab/root.go:63]: How does paging works here (I see you pass a PerPage option). Will list iterate through all pages, returning the full result ?

Comment (B3rs) [reply internal/providers/gitlab/root.go:63]: in the end it was very easy to introduce pagination so I did :) thanks for the hint!

**Review by mcorbin — APPROVED** [review]

**Pull request merged**